app/decorators/models/solidus_sitemap/spree/product_decorator.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module SolidusSitemap
+  module Spree
+    module ProductDecorator
+      module ClassMethods
+        def last_updated
+          order(updated_at: :desc).take.try(:updated_at)
+        end
+      end
+
+      def self.prepended(base)
+        base.extend ClassMethods
```

Comment:
Can you use `base.singleton_class.prepend ClassMethods` here? That way, you use `prepend` on the eigenclass as well.